app/facades/board_game_facade.rb

```diff
@@ -5,22 +5,12 @@ def self.search(name)
     data = BoardGamesService.find_board_games(name)
 
     data.map do |game|
-      categories = []
-      available_categories = BoardGamesFacade.category
```

Comment:
This was where you were getting rate limited. The facade would make a category api call for each iteration. Luckily we only need that call once. Moving this line up out of the map should solve that issue